master/buildbot/www/avatar.py

```diff
@@ -30,6 +32,29 @@ def getUserAvatar(self, email, size, defaultAvatarUrl):
         raise NotImplementedError()
 
 
+class AvatarGitHub(AvatarBase):
+    name = "github"
+    cache = {}
+
+    def getUserAvatar(self, email, size, defaultAvatarUrl):
+        url = self.cache.get(email)
+        if url:
+            raise resource.Redirect(url)
+
+        # https://developer.github.com/v3/search/#search-users
+        url = 'https://api.github.com/search/users?'
+        url += urllib.urlencode({'q': email, 'in': 'email', 'type': 'user'})
+        try:
+            res = json.loads(urllib.urlopen(url).read())
```

Comment:
Here is how I would use txrequest for this code:
```
class AvatarGitHub(AvatarBase):
    name = "github"
    cache = {}
    BASE_URL = 'https://api.github.com'  # can be overriden for GHE
    SEARCH_PATH = '/v3/search/users'

    def getSession(self):
        return txrequests.Session()

    @defer.inlineCallbacks
    def getUserAvatar(self, email, size, defaultAvatarUrl):
        url = self.cache.get(email)
        if url:
            raise resource.Redirect(url)

        search_url = self.BASE_URL + self.SEARCH_PATH
        with self.getSession() as session:
            res = yield session.get(search_url,
                                    params={'q': email, 'in': 'email', 'type': 'user'})
            if res.status_code != 200:
                raise resource.Redirect(str(defaultAvatarUrl))
            github_url = str(res.json(['items'][0]['avatar_url']))
            github_url += '&' + urllib.urlencode({'s': str(size)})
            self.cache[email] = github_url
            raise resource.Redirect(github_url)

```

You can the mock/override self.getSession in order to test everything.
Please avoid to make your test hit the internet to work mockint getSession will help to replace session.get() by a mock which emulate internet hit.